facturx/facturx.py

```diff
@@ -147,3 +147,129 @@ def write_xml(self, path):
         with open(path, 'wb') as f:
             f.write(self.xml_str)
 
+    def __make_dict(self):
+        field_file = os.path.join(os.path.dirname(__file__), 'flavors\\fields.yml')
+        req_list = []
+        with open(field_file, 'r') as f:
+            data = f.read()
+            dict_data = yaml.load(data)
+
+        for k, v in dict_data.items():
+            for kv, vv in v.items():
+                if kv == '_required':
+                    if vv is True:
+                        req_list.append(k)
+
+        factx_value = []
+        factx_key = []
+        zugf_key = []
+        zugf_value = []
+
+        for k, v in dict_data.items():
+            for item in req_list:
+                if k == item:
+                    for vk, vv in v.items():
+                        if vk == '_path':
+                            for vvk, vvv in vv.items():
+                                if vvk == 'factur-x':
+                                    factx_key.append(k)
+                                    factx_value.append(vvv)
+                                elif vvk == 'zugferd':
+                                    zugf_key.append(k)
+                                    zugf_value.append(vvv)
+
+        self.factx = {}
+
+        for i in range(len(factx_key)):
+            self.factx[factx_key[i]] = factx_value[i]
+
+        self.zugf = {}
+
+        for i in range(len(zugf_key)):
+            self.zugf[zugf_key[i]] = zugf_value[i]
+
+        self.zugf_ns = {'udt': 'urn:un:unece:uncefact:data:standard:UnqualifiedDataType:15',
+              'ram': 'urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:12',
+              'rsm': 'urn:ferd:CrossIndustryDocument:invoice:1p0',
+              'xsi': 'http://www.w3.org/2001/XMLSchema-instance'
+              }
+
+        self.factx_ns = {'xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+              'udt': 'urn:un:unece:uncefact:data:standard:UnqualifiedDataType:100',
+              'rsm': 'urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100',
+              'ram': 'urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:100',
+              'qdt': 'urn:un:unece:uncefact:data:standard:QualifiedDataType:100'
+              }
+
+    def __export_file_from_xml(self, ns, xml_file_path, json_file_path, flavor):
+        self.__make_dict()
+        json_factx = {}
+
+        # xml_file_path = os.path.join(os.path.dirname(__file__), 'text.xml')
+
+        if flavor == 'factur-x':
```

Comment:
Same as above. Can't hardcode flavors as we will have more and we can't hardcode them all. Settings for flavors need to come from the settings in the flavors class.

Reply:
Taken care of.